#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_permissions.dpatch by  <hamish@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Override PDF permission bits

@DPATCH@

--- xpdf-3.00.orig/xpdf/XPDFCore.cc
+++ xpdf-3.00/xpdf/XPDFCore.cc
@@ -4,6 +4,8 @@
 //
 // Copyright 2002-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -940,9 +942,11 @@
 // global variables (currentSelection and currentSelectionOwner).
 
 void XPDFCore::copySelection() {
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToCopy()) {
     return;
   }
+#endif
   if (currentSelection) {
     delete currentSelection;
   }
@@ -997,9 +1001,11 @@
 }
 
 GString *XPDFCore::extractText(int xMin, int yMin, int xMax, int yMax) {
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToCopy()) {
     return NULL;
   }
+#endif
   return out->getText(xMin, yMin, xMax, yMax);
 }
 
@@ -1008,9 +1014,11 @@
   TextOutputDev *textOut;
   GString *s;
 
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToCopy()) {
     return NULL;
   }
+#endif
   textOut = new TextOutputDev(NULL, gTrue, gFalse, gFalse);
   if (!textOut->isOk()) {
     delete textOut;
@@ -1647,6 +1658,7 @@
 	  core->setCursor(None);
 	  core->moveSelection(mx, my);
 #ifndef NO_TEXT_SELECT
+#ifdef ENFORCE_PERMISSIONS
 	  if (core->selectXMin != core->selectXMax &&
 	      core->selectYMin != core->selectYMax) {
 	    if (core->doc->okToCopy()) {
@@ -1655,6 +1667,9 @@
 	      error(-1, "Copying of text from this document is not allowed.");
 	    }
 	  }
+#else
+      core->copySelection();
+#endif
 #endif
 	}
 	if (core->hyperlinksEnabled) {
--- xpdf-3.00.orig/xpdf/XPDFViewer.cc
+++ xpdf-3.00/xpdf/XPDFViewer.cc
@@ -4,6 +4,8 @@
 //
 // Copyright 2002-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -2285,10 +2287,12 @@
   PSOutputDev *psOut;
 
   doc = viewer->core->getDoc();
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToPrint()) {
     error(-1, "Printing this document is not allowed.");
     return;
   }
+#endif
 
   viewer->core->setBusyCursor(gTrue);
 
--- xpdf-3.00.orig/xpdf/pdfimages.cc
+++ xpdf-3.00/xpdf/pdfimages.cc
@@ -4,6 +4,8 @@
 //
 // Copyright 1998-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -119,11 +121,13 @@
   }
 
   // check for copy permission
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToCopy()) {
     error(-1, "Copying of images from this document is not allowed.");
     exitCode = 3;
     goto err1;
   }
+#endif
 
   // get page range
   if (firstPage < 1)
--- xpdf-3.00.orig/xpdf/pdftops.cc
+++ xpdf-3.00/xpdf/pdftops.cc
@@ -4,6 +4,8 @@
 //
 // Copyright 1996-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -266,12 +268,14 @@
     goto err1;
   }
 
+#ifdef ENFORCE_PERMISSIONS
   // check for print permission
   if (!doc->okToPrint()) {
     error(-1, "Printing this document is not allowed.");
     exitCode = 3;
     goto err1;
   }
+#endif
 
   // construct PostScript file name
   if (argc == 3) {
--- xpdf-3.00.orig/xpdf/pdftotext.cc
+++ xpdf-3.00/xpdf/pdftotext.cc
@@ -4,6 +4,8 @@
 //
 // Copyright 1997-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -160,12 +162,14 @@
     goto err2;
   }
 
+#ifdef ENFORCE_PERMISSIONS
   // check for copy permission
   if (!doc->okToCopy()) {
     error(-1, "Copying of text from this document is not allowed.");
     exitCode = 3;
     goto err2;
   }
+#endif
 
   // construct text file name
   if (argc == 3) {
